tests/02_00_06-2wp-new-minimum.js

```diff
@@ -1,117 +1,130 @@
+        const federationAddressBalanceFinal = Number(await btcTxHelper.getAddressBalance(federationAddress));
+        expect(federationAddressBalanceFinal).to.be.equal(Number(federationAddressBalanceInitial + PEGIN_VALUE_IN_BTC));
     });
 
-    it('should peg-out when sending minimum value', async () => {
+    it('should peg-out successfully when sending exactly the minimum pegout value', async () => {
         // Do a peg-in first to ensure the federation has funds to do the peg-out
-        const AMOUNT_TO_LOCK_IN_BTC = 10;
-        let senderAddressInformation = await btcClient.generateBtcAddress('legacy');
+        const PEGIN_VALUE_IN_BTC = 1;
+        const MIN_PEGOUT_VALUE_IN_RBTC = 0.0025;
+        const TX_FEE_IN_RBTC = 0.001;
+
+        // Create legacy type address to use as sender
+        const senderAddressInfo = await btcTxHelper.generateBtcAddress('legacy');
+
+        await btcTxHelper.fundAddress(senderAddressInfo.address, PEGIN_VALUE_IN_BTC + btcTxHelper.getFee());
 
-        await btcClient.fundAddress(senderAddressInformation.address, AMOUNT_TO_LOCK_IN_BTC + btcClient.getFee());
+        // Get the RSK address where the funds should be locked to
+        const recipientRskAddressInfo = getDerivedRSKAddressInformation(senderAddressInfo.privateKey, btcTxHelper.btcConfig.network);
+        await rskTxHelper.importAccount(recipientRskAddressInfo.privateKey);
+        const unlocked = await rskTxHelper.unlockAccount(recipientRskAddressInfo.address);
+        expect(unlocked, 'Account was not unlocked').to.be.true;
+        const recipientRskAddressBalanceInitial = Number(await rskTxHelper.getBalance(recipientRskAddressInfo.address));
+        expect(recipientRskAddressBalanceInitial).to.be.equal(0);
 
-        const peginBtcTxHash = await _2wpUtilsLegacy.sendPegin(rskClient, btcClient, senderAddressInformation, AMOUNT_TO_LOCK_IN_BTC);
-        await _2wpUtilsLegacy.ensurePeginIsRegistered(rskClient, peginBtcTxHash);
-        let rskDerivedAddress = keyUtils.getRskAddress(senderAddressInformation.privateKey);
-        let rskPrivKey = keyUtils.privKeyToRskFormat(senderAddressInformation.privateKey);
-        await rskClient.eth.personal.importRawKey(rskPrivKey, "");
-        await rskClient.eth.personal.unlockAccount(rskDerivedAddress, '');
+        // Execute peg-in
+        const btcPeginTxHash = await sendPegin(rskTxHelper, btcTxHelper, senderAddressInfo, PEGIN_VALUE_IN_BTC);
+        await ensurePeginIsRegistered(rskTxHelper, btcPeginTxHash);
 
-        let finalDerivedAddressBalance = await rskClient.eth.getBalance(rskDerivedAddress);
-        expect(Number(finalDerivedAddressBalance)).to.equal(rsk.btcToWeis(AMOUNT_TO_LOCK_IN_BTC));
+        const recipientRskAddressBalanceAfterPegin = Number(await rskTxHelper.getBalance(recipientRskAddressInfo.address));
+        expect(recipientRskAddressBalanceAfterPegin).to.be.equal(btcEthUnitConverter.btcToWeis(PEGIN_VALUE_IN_BTC));
```

Comment:
I'd Suggest We Move The Pegin Logic To A Seperate Method, So This Test Can Focus On The Pegout Logic

Reply:
hmm.. how do i do peg-in here is little bit specific for this test case. don't think this function will be reusable for another tests. so it doesn't makes much sense for me to move it to a separate method ..